node/src/Pos/Client/CLI/Params.hs

```diff
@@ -112,14 +114,38 @@ getNodeParams cArgs@CommonNodeArgs{..} NodeArgs{..} systemStart = do
         , ..
         }
 
-getTransportParams :: CommonNodeArgs -> NetworkConfig kademlia -> TransportParams
-getTransportParams args networkConfig = TransportParams { tpTcpAddr = tcpAddr }
-  where
-    tcpAddr = case ncTopology networkConfig of
-        TopologyBehindNAT{} -> TCP.Unaddressable
-        _ -> let (bindHost, bindPort) = bindAddress args
-                 (externalHost, externalPort) = externalAddress args
-                 tcpHost = BS8.unpack bindHost
-                 tcpPort = show bindPort
-                 tcpMkExternal = const (BS8.unpack externalHost, show externalPort)
-             in  TCP.Addressable $ TCP.TCPAddrInfo tcpHost tcpPort tcpMkExternal
+data NetworkTransportMisconfiguration =
+
+      -- | A bind address was not given.
+      MissingBindAddress
+
+      -- | An external address was not given.
+    | MissingExternalAddress
+
+      -- | An address was given when one was not expected (behind NAT).
+    | UnnecessaryAddress
+
+instance Show NetworkTransportMisconfiguration where
+    show MissingBindAddress     = "No network bind address given"
+    show MissingExternalAddress = "No external network address given"
+    show UnnecessaryAddress     = "Network address given when none was expected"
```

Comment:
Should we list the actual flag names here so it's clear what we're referring to? e.g. "Network address given (by --listen or --addres) when none was expected". And same for the other two complaining about lack of flags.